packages/client/src/devices/InputMediaDeviceManager.ts

```diff
@@ -63,16 +63,23 @@ export abstract class InputMediaDeviceManager<
    * Starts stream.
    */
   async enable() {
-    if (this.state.status === 'enabled') return;
-    this.enablePromise = this.unmuteStream();
-    try {
-      await this.enablePromise;
-      this.state.setStatus('enabled');
-      this.enablePromise = undefined;
-    } catch (error) {
-      this.enablePromise = undefined;
-      throw error;
-    }
+    if (this.state.pendingStatus === 'enabled') return;
+    const signal = this.nextAbortableStatusChangeRequest('enabled');
+    const doEnable = async () => {
+      if (signal.aborted) return;
```

Comment:
5. Check if the request was canceled. That can happen if the request was scheduled to run, but then a new request came in and canceled it (see p. 2).